### PR TITLE
fix: genesis hash rate estimate

### PR DESCRIPTION
**Pull request by zhangsoledad**

### CODE REVIEW

File: spec/src/consensus.rs [hunk 1]
```diff
@@ -196,11 +196,10 @@ pub fn build_genesis_epoch_ext(
...
     let remainder_reward = Capacity::shannons(epoch_reward.as_u64() % genesis_epoch_length);
 
-    // let genesis_orphan_count = genesis_epoch_length * genesis_orphan_rate.0 / genesis_orphan_rate.1;
+    let genesis_orphan_count =
+        genesis_epoch_length * genesis_orphan_rate.0 as u64 / genesis_orphan_rate.1 as u64;
     let genesis_hash_rate = compact_to_difficulty(compact_target)
-        * genesis_epoch_length
-        * (genesis_orphan_rate.0 + genesis_orphan_rate.1)
-        / genesis_orphan_rate.1
+        * (genesis_epoch_length + genesis_orphan_count)
```
Comment (zhangsoledad) [inline spec/src/consensus.rs:202]: :fearful: :hammer:

**Review by doitian — APPROVED** [review]

File: spec/src/consensus.rs (hunk 1 above)
Comment (doitian) [reply spec/src/consensus.rs:202]: Which versions does it affect?

Comment (yangby-cryptape) [reply spec/src/consensus.rs:202]: Ref: [PR-2379: feat: let the consensus params `orphan_rate_target` to be configurable](https://github.com/nervosnetwork/ckb/pull/2379#issue-527911739)

**Review by yangby-cryptape — APPROVED** [review]

File: spec/src/consensus.rs (hunk 1 above)
Comment (yangby-cryptape) [reply spec/src/consensus.rs:202]: I still think this should be high precision computation according the description in [RFC](https://github.com/nervosnetwork/rfcs/blob/v2020.01.15/rfcs/0020-ckb-consensus-protocol/0020-ckb-consensus-protocol.md#computing-the-next-epochs-main-chain-block-number).
:tired_face: But we **CAN NOT** change it.

**Comment by zhangsoledad** [comment]
bors r=doitian,yangby-cryptape

File: spec/src/consensus.rs (hunk 1 above)
Comment (doitian) [reply spec/src/consensus.rs:202]:
```
$ git tag --contains be268a3aa
v0.39.0-rc1
v0.39.0-rc2
```

Only v0.39.0-pre versions

**Pull request merged**